nutridine/src/hooks/Meals/useMealsByQuery.js

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import brandIds from "../../constants/brandIds";
+import mockMeals from "../../constants/mockData/meals";
+
+export const useMealsByQuery = (query) => {
+    const [meals, setMeals] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+
+        const fetchData = async () => {
+            const apiKey = process.env.REACT_APP_NUTRITIONIX_API_KEY;
+            const appId = process.env.REACT_APP_NUTRITIONIX_APP_ID;
+
+            // Switch between mock data in development and API call in production
+            const IS_TESTING = process.env.REACT_APP_DEVELOPMENT_MODE !== "false";
+
+            if (IS_TESTING) {
+                // Use mock data in development
+                setMeals(mockMeals);
+                setError(null);
+                setLoading(false);
+            } else {
+                try {
+                    console.log("Fetching data with query:", query);
+                    const response = await axios.post(
+                        "https://trackapi.nutritionix.com/v2/search/instant",
+                        {
+                            query: Object.keys(brandIds).join(" ") + " " + query,
```

Comment:
I think you'll want to avoid including the `Object.keys(brandIds)` here and only send the post request if `query` is passed to the hook.

The nutritionIX API will throw an error if the query parameter is empty, and the fetchByMacros hook always wants the query field to be empty... As a workaround to get consistent results, i include all the restaurant names in the query parameter.

Your pizza-oriented results is liking coming from the fact you include all the restaurant names and the some pizza restaurant just to happens to be the first ~20 results returned from the server

Reply:
I see I see, that makes sense! Thanks for the insight. Will update that.

I thought the brandIds where there so that we only fetch from nearby restaurants.

Reply:
Just an update, that fixed it!

This page seems to work well now! let me know.

Reply:
Awesome, glad to hear!! And for clarity yeah you got it, the `brand_ids: Object.values(brandIds),` is what restricts restaurants to Victoria. Including all the names of restaurants in the `query` is a bit of a make-it-work situation